Userland/Libraries/LibLocate/LocateDb.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2021, Aron Lander <aron@aronlander.se>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <AK/Base64.h>
+#include <AK/BitCast.h>
+#include <AK/Format.h>
+#include <AK/HashMap.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/Queue.h>
+#include <AK/SinglyLinkedListWithCount.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <LibC/string.h>
+#include <LibC/unistd.h>
+#include <LibCore/File.h>
+#include <LibLocate/Types.h>
+#include <stdio.h>
+#include <sys/stat.h>
+
+namespace Locate {
+constexpr static auto locate_db_path = "/var/lib/locate/locate.db";
+
+class LocateDb {
+public:
+    LocateDb(String path, LocateDbMode mode);
+    ~LocateDb();
+
+    void write_header();
+    void write_directory(DirectoryInfo& dir_info);
+    bool verify_header();
+    OwnPtr<DirectoryInfo> get_next_directory();
+    const PermissionInfo* get_permission_info(uint32_t db_id);
+
+private:
+    long m_file_size;
+    FILE* m_file_handle;
+    LocateDbMode m_mode;
+    HashMap<uint32_t, NonnullOwnPtr<PermissionInfo>> m_path_relations;
+
+    enum class ChunkType {
+        DirStart = 1,
+        DirEnd = 2,
+        File = 3,
+        Dir = 4
```

Comment:
No need to explicitly assign all values, please add a comma to the last enum member, and I don't think we should abbreviate directory here (https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#names).

```suggestion
        DirectoryStart,
        DirectoryEnd,
        File,
        Directory,
```